include/net/sch_generic.h

```diff
@@ -742,6 +742,44 @@ static inline struct sk_buff *qdisc_peek_head(struct Qdisc *sch)
 	return qh->head;
 }
 
+static inline struct sk_buff *__qdisc_dequeue_tail(struct qdisc_skb_head *qh)
+{
+	struct sk_buff *skb = qh->tail;
+
+	if (likely(skb)) {
+		qh->tail = skb->prev;
+		qh->qlen--;
+		if (!qh->tail)
+			qh->head = NULL;
+		skb->prev = NULL;
+	}
+
+	return skb;
+}
+
+static inline unsigned int __qdisc_queue_drop_tail(struct Qdisc *sch,
+						   struct qdisc_skb_head *qh,
+						   struct sk_buff **to_free)
+{
+	struct sk_buff *skb = __qdisc_dequeue_tail(qh);
+
+	if (likely(skb)) {
+		unsigned int len = qdisc_pkt_len(skb);
+
+		qdisc_qstats_backlog_dec(sch, skb);
+		__qdisc_drop(skb, to_free);
+		return len;
+	}
+
+	return 0;
+}
+
+static inline unsigned int qdisc_queue_drop_tail(struct Qdisc *sch,
+						 struct sk_buff **to_free)
```

Comment:
Given that we are introducing a function analog to `qdisc_queue_drop()`, we should keep the parallels between the functions as true as possible. For example, `qdisc_queue_drop()` does not require parameter `struct sk_buff **to_free`. Why do we need to do so here?

Reply:
I was making it parallel to `qdisc_queue_drop_head()` which uses `to_free` as a parameter. I just thought this function would be more aligned to the function we're writing here, hope that's ok.